nhrpd: Incomplete uniqueness bit handling

zmw12306 opened the pull request:
According to https://datatracker.ietf.org/doc/html/rfc2332.html#section-5.2.3
No storage/checking of uniqueness flag. No rejection of duplicate unique registrations



Another violation: "Any attempt to register a binding between the protocol address and an NBMA address when this bit is set MUST be rejected with a Code of '14 - Unique Internetworking Layer Address Already Registered' if the replying NHS already has a cache entry for the protocol address and the cache entry has the 'uniqueness' bit set."

I didn't resolve this because it may require large modifications, including struct nhrp_cache, nhrp_cache_update_binding, caller functions to nhrp_cache_update_binding and also nhrp_handle_registration_request...

donaldsharp commented:
Please *PUT* your design comments in the actual commit instead of the PR on the website.  People look at the commits in the code not what is on the website in the future when they are trying to understand your code.

Jafaral commented:
still waiting for the comment to be addressed.

zmw12306 committed:
nhrpd: Incomplete uniqueness bit handling

According to https://datatracker.ietf.org/doc/html/rfc2332.html#section-5.2.3
No storage/checking of uniqueness flag. No rejection of duplicate unique registrations

Another violation: "Any attempt to register a binding between the protocol address and an NBMA address when this bit is set MUST be rejected with a Code of '14 - Unique Internetworking Layer Address Already Registered' if the replying NHS already has a cache entry for the protocol address and the cache entry has the 'uniqueness' bit set."
I didn't resolve this because it may require large modifications, including struct nhrp_cache, nhrp_cache_update_binding, caller functions to nhrp_cache_update_binding and also nhrp_handle_registration_request

Signed-off-by: zmw12306 <zmw12306@gmail.com>

zmw12306 commented:
> still waiting for the comment to be addressed.

done

zmw12306 commented:
> still waiting for the comment to be addressed.

Could you please take a look at the PR? Thanks!

RodrigoMNardi commented:
ci:rerun
